mm2src/coins/z_coin.rs

```diff
@@ -171,47 +185,72 @@ pub struct ZOutput {
 }
 
 impl ZCoin {
-    #[inline(always)]
-    pub fn z_rpc(&self) -> &(dyn ZRpcOps + Send + Sync) { self.utxo_arc.rpc_client.as_ref() }
+    #[inline]
+    pub fn utxo_rpc_client(&self) -> &UtxoRpcClientEnum { &self.utxo_arc.rpc_client }
 
-    #[inline(always)]
-    pub fn rpc_client(&self) -> &UtxoRpcClientEnum { &self.utxo_arc.rpc_client }
+    #[inline]
+    pub fn my_z_address_encoded(&self) -> String { self.z_fields.my_z_addr_encoded.clone() }
 
-    #[inline(always)]
-    pub fn is_sapling_state_synced(&self) -> bool { self.z_fields.sapling_state_synced.load(AtomicOrdering::Relaxed) }
+    #[inline]
+    pub fn consensus_params(&self) -> ZcoinConsensusParams { self.z_fields.consensus_params.clone() }
 
-    #[inline(always)]
-    pub fn my_z_address_encoded(&self) -> String { self.z_fields.my_z_addr_encoded.clone() }
+    #[inline]
+    pub fn consensus_params_ref(&self) -> &ZcoinConsensusParams { &self.z_fields.consensus_params }
 
-    /// Returns all unspents included currently unspendable (not confirmed)
-    async fn my_z_unspents_ordered(&self) -> UtxoRpcResult<Vec<ZUnspent>> {
-        let min_conf = 0;
-        let max_conf = i32::MAX as u32;
-        let watch_only = true;
+    #[inline]
+    pub async fn sync_status(&self) -> Result<SyncStatus, MmError<BlockchainScanStopped>> {
+        self.z_fields
+            .sync_state_connector
+            .lock()
+            .await
+            .current_sync_status()
+            .await
+    }
 
-        let mut unspents = self
-            .z_rpc()
-            .z_list_unspent(min_conf, max_conf, watch_only, &[&self.z_fields.my_z_addr_encoded])
-            .compat()
-            .await?;
+    #[inline]
```

Comment:
Thanks for the info! Should I remove `#[inline]` or maybe keep just in case we will make this public one day? :)

Reply:
> Thanks for the info! Should I remove `#[inline]` or maybe keep just in case we will make this public one day? :)

I think you can keep it since there is no negative effect with keeping it :)